app/code/core/Mage/Newsletter/Model/Resource/Subscriber.php

```diff
@@ -126,9 +126,13 @@ public function loadByCustomer(Mage_Customer_Model_Customer $customer)
 
         $select = $this->_read->select()
             ->from($this->getMainTable())
-            ->where('subscriber_email=:subscriber_email');
+            ->where('subscriber_email=:subscriber_email')
+            ->Where('store_id=:store_id');
```

Comment:
wouldn't it be better to filter by website_id instead of store_id? cause customers can be global or per-website so I think it could make more sense to filter by website_id no?

Reply:
It would be better. But there's no website_id column on the newsletter_subscriber table as far as I remember.

Reply:
oh right I'm so sorry, I was thinking about the customer_entity and not the newsletter_subscriber!

Reply:
No problem @fballiano! Thank you for your review!